lib/IDE/Refactoring.cpp

```diff
@@ -6563,24 +6601,143 @@ class AsyncConverter : private SourceEntityWalker {
     }
   }
 
+  /// Checks whether a binding pattern for a given decl can be printed inline in
+  /// an await call, e.g 'let ((x, y), z) = await foo()', where '(x, y)' is the
+  /// inline pattern.
+  const Pattern *
+  bindingPatternToPrintInline(const Decl *D, const ClassifiedBlock &Block,
+                              const ClosureExpr *CallbackClosure) {
+    // Only currently done for callback closures.
+    if (!CallbackClosure)
+      return nullptr;
+
+    auto Res = Block.paramPatternBindings().find(D);
+    if (Res == Block.paramPatternBindings().end())
+      return nullptr;
+
+    // The decl must have been bound to a single pattern.
+    auto &Patterns = Res->second;
+    if (Patterns.size() != 1)
+      return nullptr;
+
+    // We only do inline printing for patterns that bind multiple vars. Patterns
+    // that only bind a single var are printed by the main logic.
```

Comment:
Would it be more or less complicated to handle them the same way, rather than have two code paths?

Reply:
My initial feeling was that it probably wouldn't be any simpler to merge the two code paths as there's still special casing that we do for the single var case such as coalescing them. That being said, if we also want to better handle the `var` case you mention as well as the `prepareNames` logic, it might be worth it.